provider/terraform.rb

```diff
@@ -161,5 +161,25 @@ def generate_resource_tests(data)
         out_file: filepath
       )
     end
+
+    # Only going to be run once.
+    def generate_async(output_folder, _types, version_name)
+      return if @api.objects.select(&:autogen_async).empty?
+
+      prod_name = @api.name.underscore
```

Comment:
nit: `product_name` for consistency with https://github.com/GoogleCloudPlatform/magic-modules/blob/master/provider/terraform.rb#L120